src/client/executor.rs

```diff
@@ -125,10 +125,13 @@ impl Client {
                     .topology
                     .handle_application_error(
                         err.clone(),
-                        HandshakePhase::after_completion(conn),
+                        HandshakePhase::after_completion(&conn),
```

Comment:
This was technically fixed already since `after_completion` dropped the connection, but I updated this anyways to make the drop explicit in case `after_completion` is updated in the future to somehow hold on to the connection for longer than it ought to.